state_client.go

```diff
@@ -0,0 +1,176 @@
+package lndclient
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"google.golang.org/grpc"
+)
+
+// StateClient exposes base lightning functionality.
+type StateClient interface {
+	// SubscribeState subscribes to the current state of the wallet.
+	SubscribeState(ctx context.Context) (chan WalletState, chan error,
+		error)
+
+	// GetState returns the current wallet state without subscribing to more
+	// state updates.
+	GetState(context.Context) (WalletState, error)
+}
+
+// WalletState is a type that represents all states the lnd wallet can be in.
+type WalletState uint8
+
+const (
+	// WalletStateNonExisting denotes that no wallet has been created in lnd
+	// so far.
+	WalletStateNonExisting WalletState = 0
```

Comment:
Unfortunately the type and its constants are not exported in the `rpcperms` package in `lnd`. You think we should export them there instead?

Reply:
I mean, we have them defined in `lnrpc`?

Reply:
Yes, but the idea of the `lndclient` package is to use golang native types and not the clunky (and way too long named) RPC constants. Though for this use case it might be overkill to re-define them, yes.

Reply:
Ok, yeah if the idea is to not reference lnrpc at all it makes sense. It's a tiny amount of code duplication, so not a big deal. I'm okay with keeping it as is, just wanted to point out that these constants really exist already.